code/modules/power/singularity/emitter.dm

```diff
@@ -102,9 +120,26 @@
 	if(stat & (BROKEN))
 		return
 	if(src.state != 2 || (!powernet && active_power_usage))
-		src.active = 0
+		src.active = FALSE
+		src.safety = FALSE //Don't bother with the targeting laser if we're unpowered.
 		update_icon()
 		return
+	var/turf/T = get_turf(src.loc)
+	var/turf/beamtarget = get_turf(src.loc)
+	if(safety && !emagged) //Emagged emitters care not for your hugbox targeting lasers.
+		beammeupscotty:
+			for(var/dist = 0, dist <= 51, dist += 1) //Checks out to 50 tiles in the emitter's direction
+				if(T.opacity) //Is it an opaque turf (i.e. a wall)? Stop here.
+					break beammeupscotty
+				else if(T && T.contents)
+					for(var/atom/A in T.contents) //Otherwise, check the tile for any opaque atoms, like mecha or curtains
+						if(A.opacity) //We don't care about density. This means the targeting laser will still pass through dense, non-opaque atoms (like windows and blobs).
+							break beammeupscotty //You know kirk never actually said this?
+						else
+							beamtarget = T
+				T = get_step(T, src.dir)
+		src.Beam(beamtarget, icon_state = "sniper_beam", time = 2 SECONDS, maxdistance = 50) //Refreshes on process() so short duration is fine.

```

Comment:
```suggestion
	if(safety) // Being emagged will prevent the safety from being enabled
		var/list/hit = check_trajectory(get_step(src, dir), src)
		src.Beam(hit[0], icon_state = "sniper_beam", time = 2 SECONDS, maxdistance = 50) //Refreshes on process() so short duration is fine.
```
Double check this `hit` indexing because that might make the beam 0 distance